src/main/java/com/internet/shop/dao/impl/UserJdbcDaoImpl.java

```diff
@@ -0,0 +1,207 @@
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.setLong(4, user.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update user", e);
+        }
+        setRoleIds(user.getRoles());
+        addUserRoles(user);
+        return user;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        deleteUserRoles(id);
+        String query = "UPDATE users SET deleted = true WHERE user_id = ?";
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete user by ID: " + id, e);
+        }
+    }
+
+    @Override
+    public List<User> getAll() {
+        List<User> users = new ArrayList<>();
+        String query = "SELECT * FROM users "
+                + "JOIN users_roles ON users.user_id = users_roles.user_id "
+                + "JOIN roles ON users_roles.role_id = roles.role_id "
+                + "WHERE deleted = false";
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                User user = new User();
+                user.setId(resultSet.getLong("user_id"));
+                user.setName(resultSet.getString("name"));
+                user.setLogin(resultSet.getString("login"));
+                user.setPassword(resultSet.getString("password"));
+                users.add(user);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all users from DB", e);
+        }
+        for (User user : users) {
+            getUserRoles(user);
+        }
+        return users;
+    }
+
+    private void addUserRoles(User user) {
+        String query = "INSERT INTO users_roles (user_id, role_id) VALUES (?, ?)";
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(query)) {
+            for (Role role : user.getRoles()) {
+                statement.setLong(1, user.getId());
+                statement.setLong(2, role.getId());
+                statement.addBatch();
+            }
+            statement.executeBatch();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add user roles to DB", e);
+        }
+    }
+
+    private void setRoleIds(Set<Role> userRoles) {
+        String query = "SELECT * FROM roles";
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
```

Comment:
you can do that with one loop, change query to select * from roles where role_name = ? 
you will need only one loop then